README.md

```diff
@@ -1,55 +1,101 @@
+Equinox is a set of low dependency libraries that allow for event-sourced processing against stream-based stores including:
+* Snapshots
+* Caching
+* State management
+* Retrieval optimizations
+
+This allows *you* to compose the libraries into an architecture that fits your apps' needs. It does not and will not handle projections and subscriptions. Refer to [Propulsion](https://github.com/jet/propulsion) for those cases.
+
+# Table of Contents
+* [Getting Started](#getting-started)
+* [Design Motivation](#design-motivation)
+* [Features](#features)
+* [Currently Supported Data Stores](#currently-supported-data-stores)
+* [Components](#components)
+  * [Core libary](#core-library)
+  * [Serialization Support](#serialization-support)
+  * [Data Store Libraries](#data-store-libraries)
+  * [Projection Libraries](#projection-libraries)
+  * [Tools](#tools)
+  * [Starter Project Templates and Sample Applications](#starter-project-templates-and-sample-applications)
+* [Overview](#overview)
+* [Templates](#templates)
+* [Samples](#samples)
+* [Building](#building)
+* [Releasing](#releasing)
+* [FAQ](#faq)
+* [Acknowledgements](#acknowledgements)
+* [Further Reading](#further-reading)
+
+# Getting Started
+- If you want to start with code samples, [there's a simple `Counter` example using `Equinox.MemoryStore` for you](https://github.com/jet/equinox/blob/master/samples/Tutorial/Counter.fsx#L16)
+- If you are experienced with event sourcing, CosmosDB and F#, this sample is helpful - [~100 LOC end-to-end using CosmosDB](https://github.com/jet/equinox/blob/master/samples/Tutorial/Cosmos.fsx#L36) 
+- If you are experienced with CosmosDB and [CosmosStore](https://github.com/Dzoukr/CosmoStore), but unsure how Equinox can refer here - [Access Strategies guide](https://github.com/jet/equinox/blob/master/DOCUMENTATION.md#access-strategies)
+- For all new comers, keep reading!
+
+# Design Motivation
+Equinox's design is informed by discussions, talks and countless hours of hard and thoughtful work invested into many previous systems, [frameworks](https://github.com/NEventStore), [samples](https://github.com/thinkbeforecoding/FsUno.Prod), [forks of samples](https://github.com/bartelink/FunDomain), the outstanding continuous work of the [EventStore](https://github.com/eventstore) founders and team and the wider [DDD-CQRS-ES](https://groups.google.com/forum/#!forum/dddcqrs) community. It would be unfair to single out even a small number of people despite the immense credit that is due. Some aspects of the implementation are distilled from [`Jet.com` systems dating all the way back to 2013](http://gorodinski.com/blog/2013/02/17/domain-driven-design-with-fsharp-and-eventstore/).
+
+An event sourcing system usually has the following attributes:
+1. Storing events with good performance and debugging capabilities
+2. Transaction processing
+    - Optimistic concurrency
```

Comment:
perhaps add (need to load conflicting events and retry if another call overlapped on the same stream)